src/index.js

```diff
@@ -49,6 +49,7 @@ export default function onClickOutsideHOC(WrappedComponent, config) {
     constructor(props) {
       super(props);
       this._uid = uid();
+      this.document = config && config.getDocument ? config.getDocument(this.getInstance()) : global.document;
```

Comment:
You could just make `getDocument` having a default value:
```js
const defaultGetDocument = () => typeof document !== "undefined" ? document : null
export default function onClickOutsideHOC(WrappedComponent, {
  getDocument = defaultGetDocument,
  excludeScrollbar = false,
  handleClickOutside,
} = {}) {
```
and remove those ifs (`config &&`) all over the place (well, in like 3 places :P).

It is not a good practice to use `global` in library, there is no such thing and it has to be polyfilled by users to make this work - just try to avoid this if you can.

Does your `getDocument` needs an argument (`this.getInstance()`)? What for?

Reply:
The reason i have `this.getInstance()` is because only the component knows where its being rendered. Component may have some props, to identify where it should be rendered

Reply:
Seems legit. 👍

Reply:
We are using this for components/widgets that are rendered on iframe using https://github.com/ryanseddon/react-frame-component

Example

````
import listensToClickOutside from '@olasearch/react-onclickoutside'
import Frame from 'react-frame-component'

const Dropdown = listensToClickOutside(() => <div />, {
  getDocument: function (instance) {
    return instance.context.document
  }
})

Dropdown.contextTypes = {
  /* Iframe document */
  document: PropTypes.object
}

const Dropdown = listensToClickOutside(() => <div />, {
  getDocument: function (instance) {
    return instance.context.document
  }
})

ReactDOM.render(
  <Frame>
    <Chat />
  </Frame>
  , document.body
)
````

I want to avoid widgets to bind events to parent document.

Reply:
Ok, seems good. Other solution would require using `ref` on `Frame` and retrieving somehow the iframe's document, but it would be too hard to pass that to the HOC's `config`